node/src/store.rs

```diff
@@ -147,50 +145,15 @@ pub trait Store: Send + Sync + Debug {
     /// `Ok(None)` indicates that header is in the store but sampling metadata is not set yet.
     async fn get_sampling_metadata(&self, height: u64) -> Result<Option<SamplingMetadata>>;
 
-    /// Append a range of headers maintaining continuity from the genesis to the head.
-    ///
-    /// # Note
+    /// Insert a range of headers into the store.
     ///
-    /// This method does not validate or verify that `headers` are indeed correct.
-    async fn append_unchecked(&self, headers: Vec<ExtendedHeader>) -> Result<()> {
-        for header in headers.into_iter() {
-            self.append_single_unchecked(header).await?;
-        }
+    /// `verify_neighbours` determines whether entire range will be validated against headers next
+    /// to them (if present in store), or whether to just validate a contiguity of the inserted
+    /// header range
+    async fn insert(&self, headers: VerifiedExtendedHeaders) -> Result<()>;
```

Comment:
I meant the following and remove all `try_into`/`into` from callers.


```suggestion
    async fn insert(&self, headers: impl TryInto<VerifiedExtendedHeaders>) -> Result<()>;
```

Reply:
Nice! applied